### PR TITLE
A few minor changes

### PR DESCRIPTION
**Pull request by travisbrown**
There are a couple of reversions here (formatting and boxing-related), plus handling for negative zero floats, plus a couple of tweaks to the tests. Apart from these things https://github.com/circe/circe/pull/603 looks good to me.

### CODE REVIEW

File: modules/testing/shared/src/main/scala/io/circe/testing/ArbitraryInstances.scala [hunk 1]
```diff
@@ -40,7 +40,8 @@ trait ArbitraryInstances extends ArbitraryJsonNumberTransformer with CogenInstan
       arbitrary[BiggerDecimal].map(JsonBiggerDecimal(_)),
       arbitrary[BigDecimal].map(JsonBigDecimal(_)),
       arbitrary[Long].map(JsonLong(_)),
-      arbitrary[Double].map(d => if (d.isNaN || d.isInfinity) JsonDouble(0.0) else JsonDouble(d))
+      arbitrary[Double].map(d => if (d.isNaN || d.isInfinity) JsonDouble(0.0) else JsonDouble(d)),
+      arbitrary[Float].map(f => if (f.isNaN || f.isInfinity) JsonDouble(0.0f) else JsonFloat(f))
```
Comment (carymrobbins) [inline modules/testing/shared/src/main/scala/io/circe/testing/ArbitraryInstances.scala:44]: Maybe `JsonDouble(0.0f)` should actually be `JsonFloat(0.0f)` ?

**Comment by carymrobbins** [comment]
Just one small comment. Outside of that, looks good to merge.

**Comment by carymrobbins** [comment]
Also, want me to squash this all into one commit?

**Comment by travisbrown** [comment]
Ugh, yes, you're right about the `Arbitrary`. I'm not at a computer for the next hour but can fix it then, or if you don't mind you could do it.

I could go either way on squashing—your call.

**Comment by carymrobbins** [comment]
Merged and squashed, thanks!

**Pull request merged**